src/patternfly/components/Button/button.scss

```diff
@@ -370,11 +370,14 @@
       --pf-c-button--FontSize: var(--pf-c-button--m-link--m-inline--FontSize);
 
       display: inline;
-      padding: 0;
       text-align: left;
       white-space: normal;
       cursor: pointer;
 
+      &:not(.pf-m-in-progress) {
+        padding: 0;
+      }
```

Comment:
This is going to give the in-progress button padding on all sides, which I don't think we want. Also it will increase the specificity of the padding. So if someone had a local override in their app for something like adding a small left/right padding to these buttons via `.pf-c-button.pf-m-link.pf-m-inline { padding: 0 .5rem; }`, that will no longer work since the original compiled selector changed to `.pf-c-button.pf-m-link.pf-m-inline:not(.pf-m-in-progress)`, which has higher specificity than their rule. This isn't always a huge deal since we encourage using vars to override properties, but there is no var where we set `padding: 0`, and the button is a really common component, so it would be good to avoid changing the specificity if we can. Though I don't think you need to do this anyways, you could probably just add a `padding-left` with a var for the inline link progress button.

Here's a screenshot of what it looks like with this change in context of a paragraph with text around it, which is often how inline link buttons are used.

![Sep-20-2022 16-54-06](https://user-images.githubusercontent.com/35148959/191373519-6c5261d0-8d3d-4c1c-86c9-d17f8562f45f.gif)

I added this CSS to the `pf-m-in-progress` button 

```
    padding: 0; // overrides the padding added from this PR
    --pf-c-button__progress--Left: var(--pf-global--spacer--xs); // a super small left spacer to move the progress spinner to the right a bit - otherwise it looks a little too close to the text to the left of the button
    padding-left: calc(var(--pf-global--spacer--xs) + 1rem + var(--pf-global--spacer--sm)); // left padding for the spinner - xs spacer that equals the `__progress--Left` value used above + 1rem (the size of the spinner) + a sm spacer to create space between the spinner and the button text
```

This is what it looks like after

![Sep-20-2022 17-20-27](https://user-images.githubusercontent.com/35148959/191375108-3ff94d36-90f3-4004-a714-0c00ad57b143.gif)

I'm guessing something like that might work?

@mcarrano what I have above will shift any text to the right of the inline link button over a bit to make room for the spinner when it appears - as you can see in the screenshot above. Is that OK? It doesn't happen with non-inline buttons because those buttons have default left/right padding that we can shift around to make room for the spinner SVG without changing the overall width of the button. The inline link button doesn't have left/right padding by default.

Reply:
@mcoker I think that's fine. I really don't see this being used in a block of text like that anyway. Most of the places this might be used would be within a form or similar place where using a standard bordered button will be too heavy visually.